src/shared/messaging/port-provider.ts

```diff
@@ -180,27 +182,44 @@ export class PortProvider implements Destroyable {
       }
       this._handledRequests.add(requestId);
 
+      // If the source window has an opaque origin [1], `event.origin` will be
+      // the string "null". This is not a legal value for the `targetOrigin`
+      // parameter to `postMessage`, so remap it to "*".
+      //
+      // [1] https://html.spec.whatwg.org/multipage/origin.html#origin.
+      //     Documents with opaque origins include file:// URLs and
+      //     sandboxed iframes.
+      const targetOrigin = origin === 'null' ? '*' : origin;
+
       // Create the channel for these two frames to communicate and send the
       // corresponding ports to them.
       const messageChannel =
         channel === 'sidebar-host'
           ? this._sidebarHostChannel
           : new MessageChannel();
 
+      // The sidebar can only connect once. It might try to connect a second
+      // time if something causes the iframe to reload. We can't recover from
+      // this yet. Instead we just log a warning here. The port discovery
+      // protocol doesn't have a way to return errors, so the sidebar will only
+      // learn about this when it times out waiting for a response.
+      if (messageChannel === this._sidebarHostChannel) {
+        if (this._sidebarConnected) {
+          console.warn(
+            'Ignoring second request from Hypothesis sidebar to connect to host frame',
+          );
+          return;
+        } else {
+          this._sidebarConnected = true;
+        }
```

Comment:
We probably don't need the `else`, as the `if` returns early.

```suggestion

        this._sidebarConnected = true;
```

Reply:
Indeed, will do.